java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -693,6 +693,9 @@ public int hashCode(int index) {
 
     @Override
     public int hashCode(int index, ArrowBufHasher hasher) {
+      if (isNull(index)) {
```

Comment:
I don't think this is needed given the current implementation, is index would point to ZeroVector which would hash?

Reply:
I think you are right. 

However, that means there is some inconsistency in the current implementation. Maybe it is another problem that needs to be fixed.

Reply:
There is another problem here: the result of getVector(index) can be null. I have fixed this problem, and the above problem, by revising the implementation of isNull(index). 

Pls take a look.